File: src/main/java/com/imagegame/service/PlayerService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.Player;
import com.imagegame.dto.PlayerDTO;
import com.imagegame.dto.PlayerSearchDTO;
import com.imagegame.dto.PlayerPageDTO;
import com.imagegame.dto.PlayerConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PlayerService extends GenericService<Player, Integer> {

	List<Player> findAll();

	ResultDTO addPlayer(PlayerDTO playerDTO, RequestDTO requestDTO);

	ResultDTO updatePlayer(PlayerDTO playerDTO, RequestDTO requestDTO);

    Page<Player> getAllPlayers(Pageable pageable);

    Page<Player> getAllPlayers(Specification<Player> spec, Pageable pageable);

	ResponseEntity<PlayerPageDTO> getPlayers(PlayerSearchDTO playerSearchDTO);
	
	List<PlayerDTO> convertPlayersToPlayerDTOs(List<Player> players, PlayerConvertCriteriaDTO convertCriteria);

	PlayerDTO getPlayerDTOById(Integer playerId);







}

File: imagegame_ui/src/services/PlayerService.js
import http from "../http-common"; 

class PlayerService {
  getAllPlayers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/player/players`, searchDTO);
  }

  get(playerId) {
    return this.getRequest(`/player/${playerId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/player?field=${matchData}`, null);
  }

  addPlayer(data) {
    return http.post("/player/addPlayer", data);
  }

  update(data) {
  	return http.post("/player/updatePlayer", data);
  }
  
  uploadImage(data,playerId) {
  	return http.postForm("/player/uploadImage/"+playerId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PlayerService();
